agent/include/content_analysis/sdk/result_codes.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 The Chromium Authors.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CONTENT_ANALYSIS_AGENT_INCLUDE_CONTENT_ANALYSIS_SDK_RESULT_CODES_H_
+#define CONTENT_ANALYSIS_AGENT_INCLUDE_CONTENT_ANALYSIS_SDK_RESULT_CODES_H_
+
+#include <stdint.h>
+
+namespace content_analysis {
+namespace sdk {
+
+// Result codes of methods and functions.
+// Results codes with the msb set (i.e. negative numbers) are unrecoverable failures.
+enum class ResultCode : uint32_t {
```

Comment:
Should we be explicit about declaring the values' int values? If we don't then we might break it across SDK versions and cause client bugs.

Reply:
This is needed because by default enum types are signed (in MSVC) and it does not like the line `ERR_FIRST_UNCOVERABLE_ERROR = 0x80000000`.  I get a compiler warning that the value is out of range.  But I could remove this by doing `ERR_FIRST_UNCOVERABLE_ERROR = INT32_MIN` and then changing IsRecoverableError:
```
inline bool IsRecoverableError(ResultCode err) {
  return err >= ResultCode::OK;
}
```